packages/@lightningjs/ui-components/src/components/Label/Label.d.ts

```diff
@@ -18,17 +18,36 @@
 
 import lng from '@lightningjs/core';
 import Base from '../Base';
-import type { Color, StylePartial } from '../../types/lui';
+import { Color, StylePartial } from '../../types/lui';
 
-export type LabelStyle = {
+type LabelStyle = {
   backgroundColor: Color;
   paddingX: number;
   paddingY: number;
   radius: lng.Tools.CornerRadius;
   textStyle: lng.textures.TextTexture.Settings;
 };
 
-export default class Label extends Base {
+declare namespace Label {
+  export interface TemplateSpec extends Base.TemplateSpec {
+    /**
+     * text to display in label
+     */
+    title: string;
+  }
+  export interface TypeConfig extends lng.Component.TypeConfig {
+    SignalMapType: SignalMap;
+  }
+  export type SignalMap = {
+    updateBackground(): void;
+  };
+}
+declare class Label<
+  TemplateSpec extends Label.TemplateSpec = Label.TemplateSpec
```

Comment:
May be we should add TypeConfig here like
` TypeConfig extends Label.TypeConfig = Label.TypeConfig`
But I would like to defer to @anthony9187,  if that is needed all the time or not.

Reply:
yup, we need to pass the TypeConfig to the class declaration in order for the SignalMapType to get properly attached to the class. In general we should always be including a TypeConfig even if its just
```
TypeConfig extends lng.Component.TypeConfig = lng.Component.TypeConfig
```